src/targets/gpu/write_literals.cpp

```diff
@@ -28,7 +28,7 @@ void write_literals::apply(module& p) const
             }
             else
             {
-                std::string id = "@literal:" + std::to_string(n);
+                std::string id = p.name() + "@literal:" + std::to_string(n);
```

Comment:
It would be better with a colon after the module name: `std::string id = p.name() + ":@literal:" + std::to_string(n);`

Reply:
fixed